src/source/vector_tile_source.js

```diff
@@ -201,6 +208,22 @@ class VectorTileSource extends Evented implements Source {
         return this;
     }
 
+    _setLanguage(language?: ?string): this {
+        this.setSourceProperty(() => {
+            this.language = language;
```

Comment:
It seems that the `language` property is plain `string` here, but it's an Object in the `Source` interface and in the Map `setLanguage` condition

https://github.com/mapbox/mapbox-gl-js/blob/eade5b20cc9b61c35ea3b74c907579cd903f0e54/src/source/source.js#L54

https://github.com/mapbox/mapbox-gl-js/blob/eade5b20cc9b61c35ea3b74c907579cd903f0e54/src/ui/map.js#L1073-L1075

Should it be something like this?

```suggestion
            this.language[this.id] = language;
```

Or it may be the other way around and it should be the plain string in the `Source` interface and in the Map `setLanguage` condition?

Reply:
Language and worldview come as objects from the API in the TileJSON and I was just picking them to the source, but after some thought on it, it makes sense to simplify this to a string.